sale_order_line_position/models/sale_order_line.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    position = fields.Integer(readonly=True, index=True, default=False)
+    position_formatted = fields.Char(compute="_compute_position_formatted")
+
+    @api.depends("position")
+    def _compute_position_formatted(self):
+        for record in self:
+            record.position_formatted = record._format_position(record.position)
+
+    @api.onchange("sequence")
+    def _onchange_sequence(self):
+        if self.order_id.locked_positions:
+            return
+        lines = self.order_id.order_line.filtered(
+            lambda x: not x.display_type and x.sequence < self.sequence
+        )
+        self.position = len(lines) + 1
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        sale_ids = [
```

Comment:
can't we do this only on SO write of `order_line`?
Also, does it make sense to do it for all order states?
I think this is adding too much overhead at lines' creation.

Reply:
Or, can't we do this only when printing? (eg: override report handler, and/or via ctx key etc)

Reply:
Can not be done at printing time as it is displayed in the UI

Reply:
@simahawk This one has not been taken care of, should it ?

Reply:
if you can... ;)

Reply:
`can't we do this only on SO write of order_line?`
The write on the order is not called when a new line is created (At least not in the current test with create on a sale.order.line), And on sale creation neither.

`Also, does it make sense to do it for all order states?`
Yes because any sale.order.line should have a position and it should never change.

`I think this is adding too much overhead at lines' creation.`
Now with the `@api.model_create_multi` maybe not that much, on sale creation the `create` function is only called once!